### ISSUE TITLE
equuleus / wide-dhcpv6 build error

### ISSUE THREAD

**Issue by Neboer** [dd010101/vyos-jenkins#23]
Thank you for the great work and I managed to build a sagitta repo today in a virtual machine running debian bookworm. The build is almost success, all the Jenkins tasks are successful except a single task named equuleus/wide-dhcpv6 gave an error. The log is [here](https://pastebin.com/8rDv0vkU).

Since I haven't built the iso image because [another issue](https://github.com/dd010101/vyos-jenkins/issues/21#issuecomment-2165138812), I haven't try to assemble an iso image to see if it will work even without the successful build of the package.

**Comment by dd010101** [comment]
The `wide-dhcpv6` for equuleus fails for me too and then it works. I don't understand why it does work sometimes, it fails on compilation but the code is the same or should be so I don't get it.

You can try to run it until it passes as temporary workaround I guess...

**Comment by Neboer** [comment]
Okey I see. Please keep me update with this issue.

**Comment by dd010101** [comment]
There is issue where [first make breaks second make](https://salsa.debian.org/debian/wide-dhcpv6/-/blob/debian/20080615-23/debian/rules?ref_type=tags#L27-28). The first make builds `cfparse.y` and the second does only sometimes and that's why then the second make crashes since it doesn't have `cfparse.y` built. I guess the reason why only sometimes is some concurrency/race condition? I'm not sure why there are two makes who execute the same targets, seems redundant to me. I don't understand why it's happening, but I found workaround by using `-B|--always-make` to force all makes to execute build for `cfparse.y` and now I can build the `wide-dhcpv6` 10 times in row without failure - before `-B|--always-make` this would fail more than it succeeded. This is not really fix but fully working workaround so good enough for me.

I found additional [patch from Fedora](https://src.fedoraproject.org/rpms/wide-dhcpv6/blob/f40/f/wide-dhcpv6-0023-fix-the-parallel-build-fix.patch) to fix some parallel issue with `cfparse.y` build but this doesn't fix this issue. The issue persist even with the patch applied.

If anyone understands this more and does know how to fix this properly then let me know.

This requires fork, thus please use `https://github.com/dd010101/vyos-build.git` GIT repository for `wide-dhcpv6` and let me know if this fixes the issue for you.

**Comment by GurliGebis** [comment]
My guess would be that it is calling the makefile with multiple jobs as a parameter, so depending on which order things are compile, it breaks.
Just guessing, but it would explain the randomness of it all.
Can you try and change it, so the makefile is called with -j1?

**Comment by dd010101** [comment]
No, -j1 doesn't fix this I tried. It's more opposite when they DO run in parallel then it succeeds. If first make builds then the second make doesn't care since "it's already done", that's how I understand this.

**Comment by GurliGebis** [comment]
Not an expert on it either. But if that fixes the issue, then great 😃

**Comment by dd010101** [comment]
If it does - you tell me! It does for me.

**Comment by GurliGebis** [comment]
We'll see when I get to my build script 🙂

**Comment by Neboer** [comment]
Hey so what should I do to make it pass the build in Jenkins?

**Comment by dd010101** [comment]
> Hey so what should I do to make it pass the build in Jenkins?

You need to replace GIT URL in Multibranch Pipeline for wide-dhcpv6 and then run build again.

Go to Jenkins, find the wide-dhcpv6 job/package/pipeline, then go to **Configure** action, replace the GIT URL from `https://github.com/vyos/vyos-build.git` to `https://github.com/dd010101/vyos-build.git`. Save. Then select equuleus and run the build via **Build now** action.

You can also obtain fresh copy and use `./seed-jobs.sh create` this will also update all pipelines to their default version. Then you can use **Build now** action inside Jenkins for specific job/branch to build only what you want not everything.

**Comment by dd010101** [comment]
> @dd010101 Did you guys attempt to use a simple `sleep 1`?

This example is bit different. The GIT clone is done via Jenkins. Then sometimes later the Jenkins executes the build.sh, the build.sh runs dpkg-buildpackage, the dpkg-buildpackage [runs inner make](https://salsa.debian.org/debian/wide-dhcpv6/-/blob/debian/20080615-23/debian/rules?ref_type=tags#L27) this passed and then another inner make runs and this fails sometimes without the `-B|--always-make`. That's how I understand this aleast. You thinking about adding extra delay between the two inner make's would help? I didn't try that.

**Comment by GurliGebis** [comment]
@dd010101 can you submit a PR upstream? might make sense to try and push things like that to VyOS 😊

**Comment by dd010101** [comment]
I doubt they would approve my account since they deleted me on the forum 😄.
